### PR TITLE
Adds a cooldown to the slap emote.

### PR DESCRIPTION
**Pull request by SamHPurp**
This gets macro'd and spammed, filling chat with the messages. A favoured suicide method, and keeps being employed on mass on the shuttle.

Its spammy, annoying for all and 'MED RP'.... and two admins mentioned they wished for it.

:cl: Purpose2
tweak: Adds a cooldown to the *slap emote.
/ :cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -836,11 +836,11 @@
...
-				for(var/client/X in admins) 
-					if(check_rights(R_EVENT,0,X.mob)) 
-						to_chat(X, adminbfmessage) 
+				var/image/cross = image('icons/obj/storage.dmi',"bible")
+				var/adminbfmessage = "\blue [bicon(cross)] <b><font color=red>Bible Fart: </font>[key_name(src, 1)] (<A HREF='?_src_=holder;adminmoreinfo=\ref[src]'>?</A>) (<A HREF='?_src_=holder;adminplayeropts=\ref[src]'>PP</A>) (<A HREF='?_src_=vars;Vars=[UID()]'>VV</A>) (<A HREF='?_src_=holder;subtlemessage=\ref[src]'>SM</A>) ([admin_jump_link(src)]) (<A HREF='?_src_=holder;secretsadmin=check_antagonist'>CA</A>) (<A HREF='?_src_=holder;Smite=[UID()]'>SMITE</A>):</b>"
+				for(var/client/X in admins)
+					if(check_rights(R_EVENT,0,X.mob))
+						to_chat(X, adminbfmessage)
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/emote.dm]: Nothing changed here. This is Github being Github.

**Comment by taukausanake** [comment]
I like mass slapping though. Can you make it so it has a countdown and then a cooldown? Like you can slap someone X-amount of times before your hand ends up sore and you have to wait before you can slap again. Better solution if it can be implemented

**Comment by granodd** [comment]
I think it's kind of silly, though not necessarily in a funny way. If you are one of the people who do have it on a macro then I am in the mind that you definitely should not be revived after using it and filling everyone's text log. It will also keep people who do have it macro'd from accidentally themselves.

**Comment by TravellingMerchant** [comment]
As usual.. 

http://vignette3.wikia.nocookie.net/vidyavidya/images/8/81/No_fun.jpg/revision/latest?cb=20110513151255

Slapping is fun.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 1 above)
Comment (AndriiYukhymchak) [inline code/modules/mob/living/carbon/human/emote.dm]: You changed "new line" style. Nothing serious

**Comment by Taaketa** [comment]
Well if you don't want slapping limited you shouldn't want screaming limited.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 1 above)
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/emote.dm]: Must have been the editor automatically.

**Comment by Fruerlund** [comment]
Yes please 💯

**Comment by Doukhan** [comment]
Normally I wouldn't mind this sort of thing but it's just so overplayed and goddamn annoying.

**Comment by Birdtalon** [comment]
No. This needn't be changed.

**Comment by Spacemanspark** [comment]
@Birdtalon Then they spam deadchat with it.
And I mute them.

**Comment by Krausus** [comment]
> No Fun Allowed

Funny, this is exactly how griefers react when *their* preferred method of annoying other players is taken from them.

Sound-generating emotes are all supposed to have a cooldown (because it's too easy to be a nuisance with them), and anything that has an in-game effect (self-damage from the slap emote) is *also* supposed to have a cooldown. Whether this was left in its broken state as an oversight or a joke is unclear, but there's no denying that this is an anomaly; there's good reasons to fix it, regardless of how "fun" it is to spam sounds and chat messages.

**Comment by SamHPurp** [comment]
>Funny, this is exactly how griefers react when their preferred method of annoying other players is taken from them.

This is the number one response to all my PRs.

**Comment by tristan1333** [comment]
Griefers. What? Ok. Cool. Yeah. Sick. Nice. Ride-on brother. Awesome. Amazing. Great. Sweet. Neat. Funky. Fresh. Delightful. Wonderful. Brilliant.

**Comment by Lady-Luck** [comment]
I've never used a slap macro.

**Comment by Necaladun** [comment]
This is hugely fucking annoying for logs, admin spam in general and is just...stupid. Thanks!

**Comment by SamHPurp** [comment]
Fixed merge conflicts

**Comment by Kyep** [comment]
I support this.

**Comment by LightFire53** [comment]
From an admin and player perspective,  I also support this. Because the last thing I need when log diving is going through almost a hundred slap logs because people macro it

**Comment by ZomgPonies** [comment]
This should NOT be controversial. It allow people to spam chat, which should be considered a potential rule exploit. I've had ahelps be missed because of slap spam.

**Comment by Doukhan** [comment]
@ZomgPonies People hate  exploits, unless of course they're ones exploiting it, blatant lack of objectivity.
Same thing in the wave-dash and similar PRs.

**Comment by Anticept** [comment]
Fully support because of the spam.

**Comment by SamHPurp** [comment]
The fact that this is controversial highlights an issue in itself.

**Comment by Regen1** [comment]
@SamHPurp the only reason it's controversial is because fox is against it,

Going to take a guess that this PR will take ages to resolve

**Comment by taukausanake** [comment]
Make slapping deal brute damage to your hand. Also I still think the slap countdown and then cooldown after that would be better than just cooldown by itself

**Comment by Necaladun** [comment]
Ok first of all let's not be dickheads. The thumbsup/down show this is liked/disliked by 12/11 people, there's no conspiracy here. Rather than "no fun allowed" and other non-arguments,  let's go with actual pro's, con's, etc. 

Pro's for keeping: (I'm against, so correct me if I'm wrong)

It's fun for some people to commit suicide with it.
???

Con's for keeping:

It's annoying, both sound and spam wise
It's immershion breaking and overly silly
It's an overdone and played out may may

So it seems a fairly classic case to me of whats fun for some people annoys others. And in this case, it's annoying them in a spammy way. Thus, I say remove - this is a feature that annoys more people than enjoy it, and at the cost of immurshions and too much sillyness. There are plenty of other ways to commit suicide.

**Comment by Lady-Luck** [comment]
>@SamHPurp the only reason it's controversial is because fox is against it,

I think it's controversial because it's controversial.

**Comment by Lady-Luck** [comment]
#7124 A related issue. People think it's funny. But it does create spam.

As a compromise would it be acceptable to cut down on the lag spam without relying on cooldowns?

**Comment by ZomgPonies** [comment]
@Lady-Luck Then how would we know someone got slapped? I'm not sure that makes any sense. If you want to slap yourself to death, do it over 10 minutes like you would IRL. Anything else is just a coding oversight leading to a -slightly- humorous situation that grows old fast and has a lot of downsides.

**Comment by Aurorablade** [comment]
It takes some backstage agreeance to label something controversial and this qualifies due to the debate going on in this pr.

 Personally nothing wrong with a cooldown. I have seen people crit themselves with a slap macro plus the spam.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -66,7 +66,7 @@
 			else								//Everyone else fails, skip the emote attempt
 				return
 
-		if("scream", "screams")
+		if("scream", "screams", "slap", "slaps")
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/emote.dm]: Does this necessitate a longer cooldown than snapping your fingers?

**Comment by Crazylemon64** [comment]
> Makes a sound
> Has an in-game effect

Sounds like prime criterion for giving a cooldown

**Comment by Fox-McCloud** [comment]
Make this a 1 second cooldown for *slap, and it can be merged.

**Comment by Krausus** [comment]
> Make this a 1 second cooldown for *slap, and it can be merged.

... you mean the standard emote cooldown of 2 seconds, right?

**Comment by Fox-McCloud** [comment]
>... you mean the standard emote cooldown of 2 seconds, right?

No, I mean the non-standard cooldown of 1 second.

**Comment by Krausus** [comment]
Why would you give this one specific sound-generating emote a shorter cooldown than every single other sound-generating emote, including ones with *shorter* sound effects?

**Comment by Fox-McCloud** [comment]
Because it's what we discussed and found most appropriate for the emote.

**Comment by Krausus** [comment]
Who's "we", and what's the reasoning behind giving one emote a halved cooldown time?

**Comment by Fox-McCloud** [comment]
The maintainers and headmins. And the reasoning was we felt it fit the emote better as a cooldown than the standard 2 seconds; nothing particularly complicated.

**Comment by Krausus** [comment]
So, no particular reason behind the decision, it just "feels" like a better fit.

@Crazylemon64 @FreeStylaLT @Necaladun @TullyBurnalot, you all really feel this emote needs a unique, halved cooldown?

**Comment by SamHPurp** [comment]
This needs its own snowflaked cooldown? Forgive the outburst, but thats fucking insane.

**Comment by ZomgPonies** [comment]
So your meme emote with it's unintended behavior gets it's own special treatment that is completely different from the standard that was set for all other emotes, sounds and animations that can be spammed? (Alien flip being the most recent one)?

**Comment by Anticept** [comment]
I'm really confused by why it would get one second, when it's one of the most annoying emotes out there. I'd love to hear why it "feels" better as one second too... but I'll remain open minded about it.

**Comment by Necaladun** [comment]
This was discussed and 1 second was seen as a compromise between people who wanted none and people who wanted 2 basically. No giant conspiracy. No need for outbursts. Just a case of "some want 0 some want 2 lets go with in between."

**Comment by SamHPurp** [comment]
Hyperbole incoming: Good thing people don't compromise between death penalty and no-death penalty by selective dismemberment.

**Comment by Spartan6** [comment]
This wasn't some union of meetings deciding the fate of a country and needing compromises, this is a slap emote that spams chat.

Why are we letting it be special snowflake?

**Comment by Krausus** [comment]
> This was discussed and 1 second was seen as a compromise between people who wanted none

Why are we compromising with people whose best justification is basically "it's funny, because it annoys other people"? Should we compromise with the people who want to indiscriminately kill other players, too?

> and people who wanted 2 basically

Five. 5 seconds. This PR sets the cooldown to 5 seconds, alongside \*scream. This compromise wouldn't be so ridiculous if not for someone deciding to move the goal posts to favor one side.

> No giant conspiracy. No need for outbursts.

Neca, we're not stupid. There's a very short list of people who were opposed to this *and* in a position to decide that we need a compromise that misrepresents the people in favor... and look what we got: a "compromise" that's a 1-second slap in the face of the people who just want this emote to meet the standards we've already set. It doesn't take a conspiracy theorist to come to the conclusion that the decision making process of the "maintainers and headmins" leans conspicuously in favor of one person with a chip on their shoulder, and it should come as no surprise that people are frustrated by that.

**Comment by Krausus** [comment]
> Going to take a guess that this PR will take ages to resolve, even when it's a complete nobrainer

It really is sad that our "leadership" is this predictably dysfunctional.

**Comment by TullyBurnalot** [comment]
To address some concerns:

1) There seems to be an undercurrent remarking on Fox being the main opposition force behind this supposed "controversy". I really have no clue where you're getting this from, but the discussion the Maintainers and Headmins had about this lasted about a minute once the PR went up, and the 1 second decision was pretty much entirely a Headmin compromise, not a Maintainer one;

2) Yes, this was LITERALLY a compromise. PR was @mentioned, we all talked, someone mentioned 2 seconds, someone suggested we half that and make it 1 just so that the *slap cannot be spammed, and everyone agreed on it. It baffles me how this was somehow turned into more fuel for the fire. Is Occam's Razor lost on this Git?

3) @SamHPurp Your concept is solid and we want a cooldown on slapping. However, your steadfast refusal to change and subsequent stonewalling of what was, in essence, a _Headmin_ decision on balance and server direction, will lead to this PR being closed. You may re-open it at a later date if you're ready to make it a 1 second cooldown.

**Comment by Krausus** [comment]
> There seems to be an undercurrent remarking on Fox being the main opposition force behind this supposed "controversy". I really have no clue where you're getting this from

It's funny that you refer to this as a "supposed controversy", when you wouldn't have even had to step in had this not been tagged, by Fox, as "controversial" because... there were a bunch of down-thumbs? No, wait, we don't base PR decisions on those, so it must have been all the people bringing such compelling arguments as "no fun allowed" and... that it's okay because slappers eventually die.

Supposed controversy, indeed.

> Yes, this was LITERALLY a compromise.
> someone mentioned 2 seconds
> someone suggested we half that and make it 1
> It baffles me how this was somehow turned into more fuel for the fire.

You didn't compromise with the people in favor of this PR's 5 seconds, you compromised with the people who wanted 0 and "someone" who pulled 2 seconds out of their bum. Demanding a one-of-a-kind 1-second cooldown feels like the sort of spiteful "compromise" that you make when you'd be insane to not concede, but still want to inconvenience the person you disagree with.

> your steadfast refusal to change and subsequent stonewalling of what was, in essence, a *Headmin* decision on balance and server direction, will lead to this PR being closed. You may re-open it at a later date if you're ready to make it a 1 second cooldown.

So it doesn't matter whether the compromise made any sense, because you've made your decision and, as the highest authority on the server, that decision is final. That's your prerogative, I suppose. Thank you for showing us to never question the will of the headmins.

**Comment by Necaladun** [comment]
Please knock of the sarcasm and ridiculousness of this. Try to communicate without being so over the top. Whether we merge it with 1, 2, 5, 10 seconds, or not at all, we have to make a decision that some people will not be in favor of. 

The major aim of this was to stop slap being able to be spammed with no delay, mainly due to the chatlog spam and it being a silly method of suicide. Something I'm well in favour of. 5 seconds seems way to long, especially compared to snapping, etc.

It seemed reasonable that it's akin to flip, fart, snap, cough, etc, which are 2 seconds. Scream is 5 due to it being a louder and longer sound. Because it was more of the former and not the latter, we decided to go with 1 second, half way between those and 0. 

There's nothing preventing this from being increased later, whether to 2 seconds, 10 seconds, 1 hour, whatever. We thought a reasonable compromise between what people wanted was the best way to merge this quickly and see how it works. Sadly, due to childish stubbornness, we don't have that code yet, which is a shame, as I would love to see a delay on this.

You're free to question the "will of the headmins". Try to do so in a productive way without such hyperbole. Just because we don't always agree with you or aren't willing to argue with you indefinitely until things are changed the way YOU want to it to be, doesn't mean you can't question it.

**Comment by SamHPurp** [comment]
Everything @Krausus has said is accurate, and reflects my own thoughts.

**If the slap emote wasn't currently in the game, and instead was being merged in; the conversation would be between 2 and 5.**

This is the problem the BBC have in the UK. In their attempt to be balanced they go too far the wrong way. If they have on a climate scientist, then they invite on someone who thinks the sky is a carpet painted by God.

> Sadly, due to childish stubbornness, we don't have that code yet, which is a shame, as I would love to see a delay on this. - @Necaladun 

Yes, standing up for what you believe in is childish. Personal attacks undermine your own arguments.

There is nothing stopping anyone else from submitting a rival PR for 1 second. As I've stated before, I will be no part of a further degradation of standards. Two seconds is where it should be, and where I'd update this, if I knew doing so would get it merged.

**Comment by Krausus** [comment]
> Try to communicate without being so over the top.

You're asking a lot of me here, Neca.

> Whether we merge it with 1, 2, 5, 10 seconds, or not at all, we have to make a decision that some people will not be in favor of.

Anything non-0 will disappoint the people who want to spam it, but only non-standard cooldowns make everyone else wonder what you were thinking.

> 5 seconds seems way to long, especially compared to snapping, etc.
> It seemed reasonable that it's akin to flip, fart, snap, cough, etc, which are 2 seconds. Scream is 5 due to it being a louder and longer sound. Because it was more of the former and not the latter, we decided to go with 1 second, half way between those and 0.

This is the least hand-wavey explanation we've gotten in **three weeks.** You decided that a standard 2-second cooldown was most appropriate for this emote (which seems perfectly reasonable), then decided to use that as the base for a compromise. But it's still not clear what that compromise is meant to accomplish. Slapping *is* louder than some other 2-second emotes, and even 1 second shuts down the fun spam some people want to keep. I can't tell who is meant to benefit from this compromise. Nobody else seems to be arguing in favor of it.

> There's nothing preventing this from being increased later, whether to 2 seconds, 10 seconds, 1 hour, whatever.

So you've decided that a standard 2 seconds is unacceptable, to the point that this PR had to be closed for disobedience, but making it 2 seconds later is fine. Presumably, only after we find a better reason for 2 seconds than "that's the standard minimum used for all sound-making emotes", since that's not good enough for now.

> Sadly, due to childish stubbornness, we don't have that code yet

Wow.

> Just because we don't always agree with you or aren't willing to argue with you indefinitely until things are changed the way YOU want to it to be, doesn't mean you can't question it.

I *can* question whatever I want, but if my arguments fall flat against the decision of the headmins, there's no point. Your justification for 1 second is that that's the compromise you decided, and our justifications for 2 seconds can't compete with that. Questioning it was fruitless.

Oh, and if you want us to think we can question your decision, please don't get Fox to deliver it as a blatant ultimatum. In fact, I recommend you avoid using him as a spokesperson at all.

**Comment by tristan1333** [comment]
Wowee brother. Inconsistency and loud memes are what I'd argue make this game. We triple AAA EA Games now bois. Argue over new features not meaningless shit. If it gets merged, all good. If not, bad luck bro. Not like you put hours of work into getting a feature together. Leave our inconsistencies to the failed remakes.

**Comment by Regen1** [comment]
I feel the same

The decision on "1 second" is literally because "that seems ok with me"

**Comment by ZomgPonies** [comment]
@tristan1333 Arguments like that make it seem like you'd be more comfortable playing on Hippie than Paradise.

**Comment by tristan1333** [comment]
Negative. Good/meaningful interactions alongside what I said are what make Paradise nice. It leans every which way rather than going too deep into either.

**Comment by Anticept** [comment]
My personal thoughts on issues I'm seeing:

First: One of the root causes of friction that we're seeing is the way reasons for a decision are being framed. If the reason for this decision is a result of "this is what I want", then please, just say that. We can all see that's what's going on here. I personally have no problem with someone or someone(s) making the executive decision to steer the server's design (something that is actually a good thing). What personally upsets me is when it is put behind newspaper thin veils and cheap reasons; please just have the courtesy of saying "I personally like x, and will vote accordingly". It will still get flak for it going against the popular opinion, but at least I know where we stand, instead of the smoke and mirrors.

Now, as for the slap emote. I personally don't mind it being 1 second, AS LONG AS THE OPTION FOR FURTHER INCREASES ARE STILL ON THE TABLE. Otherwise, it won't feel like a "compromise" to me, but rather attempting to give "just enough" to placate the masses. It feels like it was on the level of a political promise.

That's how I personally feel about all this. Whether I'm wrong or not on these issues is irrelevant, it won't change how I feel about this PR discussion turned out, or how I'm feeling like github issues are being (mis)handled. I know good intentions are there. That's why I'm still around and still trying to help where I can and make things better. But, if a decision is going to be made on personal interest, then frame it as such, it will certainly go a long way.

**Comment by taukausanake** [comment]
> vote > veto

We are not a Democracy remember.

Yeah, if nothing else is going to happen to this it might as well be one second. Arguing isn't going anywhere else. Might as well be to concede and make a suggestion at a later date. You know, a better idea if one comes up

**Comment by Anticept** [comment]
@taukausanake never said we were. However, the way that PRs are approved is a headmin/maintainer vote. That's what "vote" I'm talking about.

**Comment by SamHPurp** [comment]
@TullyBurnalot Now that people have had a chance to cool off. Can this get re-opened with the aim of 2 seconds like all other emotes of this type?

**Pull request closed without merging**